add pharo 64bits linux and some minor enhancements (2nd attempt)

estebanlm opened the pull request:
this add 64bits for linux but also adapts command line to accept "pharo format" command (double dash instead one: --) 
also it includes a couple of minor PharoVM changes (like XInitThreads)

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo

estebanlm committed:
temporary disable SDL2DisplayPlugin

estebanlm committed:
build for 64bits linux

estebanlm committed:
add missing editpharoinstall.sh

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo

estebanlm committed:
update to work also as a submodule

estebanlm committed:
add reaction to shift on mac

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo

estebanlm committed:
add formatting

estebanlm committed:
added XInitThreads() for PharoVM

estebanlm committed:
added Pharo "double dash" options style

estebanlm committed:
fix bad enclosing quotes

estebanlm committed:
remove bad commited file

estebanlm committed:
remove more wrongly commited files

estebanlm committed:
remove a test macro

estebanlm committed:
grrr... yet another wrong macro

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo

# Conflicts:
#	build.linux32x86/pharo.cog.spur/plugins.ext

estebanlm committed:
add a first version of building third-party libraries for linux 32bits

estebanlm committed:
linux 64bits with dependencies working.

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo

estebanlm committed:
prepared to build linux with itimer

estebanlm committed:
Merged branch merge-with-pharo into merge-with-pharo

estebanlm committed:
prepared to build linux 64bits with itimer

estebanlm committed:
added HEARTBEAT variable to build also those VMs

estebanlm committed:
adding OPENSSL_LIBRARIES to try to make it compile.

estebanlm committed:
add third-party libs for ARMv6

estebanlm committed:
Merged branch merge-with-pharo into merge-with-pharo

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo-win32

estebanlm committed:
adding brew openssl directory

estebanlm committed:
trying to use osx standard libssl (it finds it in my machine)

j4yk commented on build.linux32x86/third-party/libgit2/Makefile.lib:
```diff
@@ -0,0 +1,52 @@
+
+# NOTE: I'm adding this ugly hack of "sed..." because CMake does not find properly libssl and libcrypto 
+# on 64bits envirenments (it always find the 64bits version, regardless the architecture).
+# this is ugly, but the only way I fond :( 
```
Did you try https://cmake.org/cmake/help/v3.0/prop_gbl/FIND_LIBRARY_USE_LIB64_PATHS.html ?

estebanlm replied:
well, this problem is to build a 32bits vm in a 64bits so I would need FIND_LIBRARY_USE_LIB32_PATHS... but anyway no, I didn't. I will try it, hopefully I can remove that ugliness :)

estebanlm replied:
btw, this is fixed because now I'm also building openssl :S

estebanlm committed:
add openssl dependency
fixed install_name_tool rewrites for all libraries
(32bits... if this passes, I will do same for 64bits)

estebanlm committed:
prepare macox64x64 to build openssl too

estebanlm committed:
update cairo libs (an attempt to fix the build... which I doubt, but worth trying)

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo

# Conflicts:
#	scripts/updateSCCSVersions

estebanlm committed:
Merge branch 'merge-with-pharo-win32' into merge-with-pharo

estebanlm committed:
add curl to rpi prerequisites

estebanlm committed:
Merge branch 'merge-with-pharo' of github.com:estebanlm/opensmalltalk-vm into merge-with-pharo

estebanlm committed:
making build of dependencies optional for rpi because they take too much time

estebanlm committed:
Merged branch merge-with-pharo into merge-with-pharo

estebanlm committed:
shifting command arguments (to handle product name as before)

estebanlm committed:
Merge branch 'Cog' of github.com:OpenSmalltalk/opensmalltalk-vm into merge-with-pharo

estebanlm commented:
I will integrate this: errors in v3 are not related to my changes (Cog branch is failing with this problem too) and the pharo errors is some random write error I asked travis support because is something I cannot handle my self (if you have an idea, please tell me :)
